src/mococrw/dilithium.h

```diff
@@ -0,0 +1,392 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2022 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+#include "mococrw/asymmetric_crypto_ctx.h"
+#include "mococrw/key.h"
+
+namespace mococrw
+{
+/*
+ * The idea is to implement dilithium in a way that we can later alias the classes here to the ones
+ * in key.h and asymmetric_crypto_ctx.h with reduced efforts in code adaptions.
+ */
+
+/**
+ * This class is a replacement of EVP_PKEY as long as dilithium is not supported by the openssl
+ * version used by MoCOCrW. It stores the key as std::vector<uint8_t> and the key type.
+ * Addittionally some convenient functions are added.
+ */
+class DilithiumKeyImpl
+{
+public:
+    enum class DilithiumParameterSet : int { DILITHIUM2, DILITHIUM3, DILITHIUM5 };
+
+    DilithiumKeyImpl(const std::vector<uint8_t> key_data,
```

Comment:
Probably this class wants a destructor that invokes the secure vector delete from our utilities on the key_data.